### PR TITLE
Remove type hinting check in transforms

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

The type hinting check is removed for transforms, in order to make the transform decorator more practical.

**Description of the Change:**

Remove the type hinting check that would fail if the user does not respect the transform contract.

The signature check between the transform and the expand transform is kept.

**Benefits:**

More practical for advanced users

**Possible Drawbacks:**

Potentially harder for beginners to create  if the contract is not understood.

### CODE REVIEW

**Comment by rmoyard** [comment]
[sc-47760]

File: pennylane/transforms/core/transform.py
Comment (timmysilv) [inline pennylane/transforms/core/transform.py:1]: lines 45 and 65 still state that the developer must provide type hints, so we might want to re-word it. Maybe we should just describe it in plain english? something like "The transform must accept a single positional argument called `tape`, and return a sequence of tapes along with a post-processing function"

Comment (rmoyard) [reply pennylane/transforms/core/transform.py:1]: Thanks I will update that :+1:

Comment (rmoyard) [reply pennylane/transforms/core/transform.py:1]: Let me know if it reads better

**Review by timmysilv — APPROVED** [review]
minimal and perfect! :shipit:

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -326,6 +326,9 @@
 
 <h4>Other improvements</h4>
 
+* The type hinting check in the `qml.transform` decorator is removed.
```
Comment (josh146) [inline doc/releases/changelog-dev.md:329]:
```suggestion
* The formal requirement that type hinting be providing when using
  the `qml.transform` decorator has been removed. Type hinting can still
  be used, but is now optional. Please use a type checker such as
  [mypy](https://github.com/python/mypy) if you wish to ensure types are
  being passed correctly.
```

**Review by josh146 — APPROVED** [review]
Nice, thanks Romain! 💯
